superset/connectors/sqla/models.py

```diff
@@ -702,13 +702,7 @@ def data(self) -> Dict[str, Any]:
             data_["fetch_values_predicate"] = self.fetch_values_predicate
             data_["template_params"] = self.template_params
             data_["is_sqllab_view"] = self.is_sqllab_view
-            # Don't return previously populated health check message in case
-            # the health check feature is turned off
-            data_["health_check_message"] = (
-                self.health_check_message
-                if config.get("DATASET_HEALTH_CHECK")
-                else None
-            )
+            data_["health_check_message"] = self.health_check_message
```

Comment:
```suggestion
            data_["health_check_message"] = self.check_health()
```


If this will run an external function every time this line is visited, maybe it's better to change this property to an actual function call so it "looks" more expensive...

Reply:
@ktmud I thought about that. I think we could also just use `@functools.lru_cache` for the property so at least the property is cached locally.